nvflare/app_common/executors/in_process_client_api_executor.py

```diff
@@ -0,0 +1,212 @@
+        self._data_bus.subscribe([TOPIC_LOG_DATA], self.log_result_callback)
+        self.local_result = None
+        self._fl_ctx = None
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.START_RUN:
+            super().handle_event(event_type, fl_ctx)
+            self._engine = fl_ctx.get_engine()
+            self._fl_ctx = fl_ctx
+            self._init_converter(fl_ctx)
+
+            self._task_fn_thread = threading.Thread(target=self._task_fn_wrapper.run)
+            self._task_fn_thread.start()
+
+        elif event_type == EventType.END_RUN:
+            self._event_manager.fire_event(TOPIC_STOP, "END_RUN received")
+            if self._task_fn_thread:
+                self._task_fn_thread.join()
+
+    def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
+        self.log_info(fl_ctx, f"execute for task ({task_name})")
+        try:
+            fl_ctx.set_prop("abort_signal", abort_signal)
+
+            meta = self._prepare_job_meta(fl_ctx, task_name)
+            client_api = InProcessClientAPI(job_metadata=meta, result_check_interval=0.5)
+            client_api.init()
+            self._data_bus.put_data(CLIENT_API_KEY, client_api)
+
+            shareable.set_header(FLMetaKey.JOB_ID, fl_ctx.get_job_id())
+            shareable.set_header(FLMetaKey.SITE_NAME, fl_ctx.get_identity_name())
+            if self._from_nvflare_converter is not None:
+                shareable = self._from_nvflare_converter.process(task_name, shareable, fl_ctx)
+
+            self.log_info(fl_ctx, "send data to peer")
+
+            self.send_data_to_peer(shareable, fl_ctx)
+
+            # wait for result
+            self.log_info(fl_ctx, "Waiting for result from peer")
+            while True:
+                if abort_signal.triggered:
+                    # notify peer that the task is aborted
+                    self._event_manager.fire_event(TOPIC_ABORT, f"{task_name}' is aborted, abort_signal_triggered")
+                    return make_reply(ReturnCode.TASK_ABORTED)
+
+                if self.local_result:
+                    result = self.local_result
+                    self.local_result = None
+                    if self._to_nvflare_converter is not None:
+                        result = self._to_nvflare_converter.process(task_name, result, fl_ctx)
+                    return result
+                else:
+                    self.log_debug(fl_ctx, f"waiting for result, sleep for {self._result_pull_interval} secs")
+                    time.sleep(self._result_pull_interval)
+
+        except Exception as e:
+            self.log_error(fl_ctx, secure_format_traceback())
+            self._event_manager.fire_event(TOPIC_ABORT, f"{task_name}' failed: {secure_format_traceback()}")
+            return make_reply(ReturnCode.EXECUTION_EXCEPTION)
+
+    def _prepare_job_meta(self, fl_ctx, task_name):
```

Comment:
The `ConfigKey.TASK_NAME: task_name,` is not a job meta.
One job contains multiple tasks.

So suggest rename this method to "_prepare_task_meta"